exonum-java-binding/core/src/main/java/com/exonum/binding/core/storage/database/Fork.java

```diff
@@ -42,6 +42,16 @@
    * to cancel it on peer ownership transfer, happening in {@link #intoPatch()}.
    */
   private final ProxyDestructor destructor;
+  /**
+   * A cleaner for this fork.
+   */
+  private final Cleaner cleaner;
+  /**
+   * A cleaner for objects depending on the fork. A separate cleaner is needed to be able to destroy
+   * the objects depending on the fork when it is converted into patch and invalidated or
+   * rolled-back (which requires collection invalidation).
+   */
+  private Cleaner forkCleaner;
```

Comment:
Yes, the name is ambiguous. collectionsCleaner/indexesCleaner are fine, but I didn't want to specify collections/indexes because this cleaner is not restricted to them.

Reply:
Any other ideas for the field names?

Reply:
`childrenCleaner` is the only one I came up with.

Reply:
![изображение](https://user-images.githubusercontent.com/7531824/65796246-c5a6a700-e174-11e9-9eb9-b929a11c2ddf.png)

Reply:
Let them be forkCleaner (of the fork) and indexCleaner (of indexes and any dependent objects).